src/config.ts

```diff
@@ -110,6 +110,28 @@ export const getApiKeyWhitelistWithFallback = (
     return result.sort();
 };
 
+export const getApiKeyFromLabel = (
+    label: string,
+): string | undefined => {
+    if (process.env.API_KEYS_ACL === undefined) {
+        return undefined;
+    }
+    let deserialized: ApiKeyStructure;
+    try {
+        deserialized = JSON.parse(process.env.API_KEYS_ACL);
```

Comment:
Is there a way we could avoid paying the deserialization cost on every call to this function? 

Maybe just make this object value an exported constant or something

Reply:
We could actually generate two lookup maps at start: 

- One by api key (which is the default)
- Another map by the label value

Reply:
that's a good call, I think we'd only call this function once on startup tho